### PR TITLE
Git: Slightly refactor configuring JSch

### PR DESCRIPTION
**Pull request by sschuberth**
Override (the by default empty) configureJSch() which gets called by
createDefaultJSch() to further configure JSch instead of overriding
createDefaultJSch() itself. Also, do not set non-host-specific
configuration on the session, but globally on JSch.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

### CODE REVIEW

File: downloader/src/main/kotlin/vcs/Git.kt [hunk 1]
```diff
@@ -59,30 +58,32 @@ const val GIT_HISTORY_DEPTH = 50
...
+                    // The default JSch instance uses identity files named "identity", "id_rsa" or "id_dsa" from the
+                    // current user's ".ssh" directory, see https://www.codeaffine.com/2014/12/09/jgit-authentication/.
 
-                override fun createDefaultJSch(fs: FS): JSch {
-                    val jSch = super.createDefaultJSch(fs)
+                    // Accept unknown hosts.
+                    JSch.setConfig("StrictHostKeyChecking", "no")
+
+                    // Limit to "publickey" to avoid "keyboard-interactive" prompts.
+                    JSch.setConfig("PreferredAuthentications", "publickey")
```
Comment (mnonnenmacher) [inline downloader/src/main/kotlin/vcs/Git.kt:73]: If the above two settings are global they could be moved out of the session factory to the top of the `init` block.

File: downloader/src/main/kotlin/vcs/Git.kt [hunk 2]
```diff
@@ -59,30 +58,32 @@ const val GIT_HISTORY_DEPTH = 50
...
             val sessionFactory = object : JschConfigSessionFactory() {
-                override fun configure(hc: OpenSshConfig.Host, session: Session) {
-                    session.setConfig("StrictHostKeyChecking", "no")
-                }
+                @Suppress("EmptyFunctionBlock")
+                override fun configure(hc: OpenSshConfig.Host, session: Session) {}
+
+                override fun configureJSch(jsch: JSch) {
+                    // The default JSch instance uses identity files named "identity", "id_rsa" or "id_dsa" from the
+                    // current user's ".ssh" directory, see https://www.codeaffine.com/2014/12/09/jgit-authentication/.
```
Comment (mnonnenmacher) [inline downloader/src/main/kotlin/vcs/Git.kt:67]: I assume this comment is there to explain the behavior if no SSH-Agent is available? If so, maybe merge it with the SSH-Agent comment below.

File: downloader/src/main/kotlin/vcs/Git.kt (hunk 1 above)
Comment (sschuberth) [reply downloader/src/main/kotlin/vcs/Git.kt:73]: Correct. I decided to still keep them here as also [JGit itself](https://github.com/eclipse/jgit/blob/231c44d553d9868795eafd9c8d87a2d3f7de50bd/org.eclipse.jgit/src/org/eclipse/jgit/transport/JschConfigSessionFactory.java#L388-L396) calls `JSch.setConfig()` in `createDefaultJSch()`, not in `configureJSch()`.

File: downloader/src/main/kotlin/vcs/Git.kt (hunk 2 above)
Comment (sschuberth) [reply downloader/src/main/kotlin/vcs/Git.kt:67]: The comment is there to explain which kind of authentication JGit configures JSch for by default. I've nevertheless merged with the comment below.

**Comment by sschuberth** [comment]
Merging despite the unrelated Travis failure.

**Pull request merged**